File: 78-subsets/78-subsets.py
class Solution:
    # Time Complexity: O(n*2^n)
    # Space Complexity O(2^n)
    def subsets(self, nums: List[int]) -> List[List[int]]:
        queue = [[]]
        # Iterate through nums
        for num in nums:
            # Make a copy of queue and iterate through
            for curr in [*queue]:
                # Append modified set into the queue
                queue.append([*curr, num])
        return queue

File: 78-subsets/78-subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
    // Time Complexity: O(n*2^n)
    // Space Complexity O(2^n)
    const queue = [[]];
    // Iterate through nums
    for (let num of nums){
        // Make a copy of queue and iterate through
        for (let curr of [...queue]){
            // Append modified set into the queue
            queue.push([...curr, num]);
        }
    }
    return queue;
};